### PR TITLE
test(e2e): Add test for `boundary version`

### PR DESCRIPTION
**Pull request by moduli**
During 0.12.0 testing, there was an issue observed where some additional characters were added to the `boundary version` output. This PR adds an e2e test to validate that output.

Here are some examples where the test would fail.
```
❯ go test -v github.com/hashicorp/boundary/testing/internal/e2e/tests/static -count=1 -run TestCliVersion
=== RUN   TestCliVersion
    version_test.go:30: [0a 12 0+oss]
    version_test.go:35: 
                Error Trace:    /Users/mycow/Developer/boundary/testing/internal/e2e/tests/static/version_test.go:35
                Error:          Received unexpected error:
                                strconv.Atoi: parsing "0a": invalid syntax
                Test:           TestCliVersion
                Messages:       Invalid value in version (0a.12.0+oss). EXPECTED: Number, ACTUAL: 0a
--- FAIL: TestCliVersion (0.10s)
```

```
❯ go test -v github.com/hashicorp/boundary/testing/internal/e2e/tests/static -count=1 -run TestCliVersion
=== RUN   TestCliVersion
    version_test.go:30: [0 12 0+oss]
    version_test.go:35: 
                Error Trace:    /Users/mycow/Developer/boundary/testing/internal/e2e/tests/static/version_test.go:35
                Error:          Received unexpected error:
                                strconv.Atoi: parsing "0+oss": invalid syntax
                Test:           TestCliVersion
                Messages:       Invalid value in version (0.12.0+oss). EXPECTED: Number, ACTUAL: 0+oss
--- FAIL: TestCliVersion (0.10s)
```

```
❯ go test -v github.com/hashicorp/boundary/testing/internal/e2e/tests/static -count=1 -run TestCliVersion
=== RUN   TestCliVersion
    version_test.go:30: [0 12 0 1]
    version_test.go:31: 
                Error Trace:    /Users/mycow/Developer/boundary/testing/internal/e2e/tests/static/version_test.go:31
                Error:          Not equal: 
                                expected: 3
                                actual  : 4
                Test:           TestCliVersion
                Messages:       Incorrect number of parts in version (0.12.0.1). EXPECTED: 3, ACTUAL: 4
--- FAIL: TestCliVersion (0.11s)
```

### CODE REVIEW

File: testing/internal/e2e/tests/static/version_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	var versionResult version.Info
+	err := json.Unmarshal(output.Stdout, &versionResult)
+	require.NoError(t, err)
+
+	parts := strings.Split(versionResult.Version, ".")
+	assert.Equal(t, 3, len(parts), "Incorrect number of parts in version (%s). EXPECTED: 3, ACTUAL: %d", versionResult.Version, len(parts))
+
+	for _, v := range parts {
+		_, err := strconv.Atoi(v)
+		require.NoError(t, err, "Invalid value in version (%s). EXPECTED: Number, ACTUAL: %s", versionResult.Version, v)
```
Comment (johanbrandhorst) [inline testing/internal/e2e/tests/static/version_test.go:36]: Maybe use assert here so it doesn't bail out on the first error but checks the rest of the parts too?

Comment (johanbrandhorst) [reply testing/internal/e2e/tests/static/version_test.go:36]: Also you can use `%q` if you want to quote it

**Review by johanbrandhorst — APPROVED** [review]

**Review by johanbrandhorst — APPROVED** [review]
